### ISSUE TITLE
ExporttoDatabase Through Python Package Error

### ISSUE THREAD

**Issue by jenna-tomkinson** [CellProfiler/CellProfiler#4646]
**Describe the bug**
Following the steps given in the [CellProfiler as a Python package](https://github.com/CellProfiler/CellProfiler/wiki/CellProfiler-as-a-Python-package) section of the Wiki, I have come to an issue when running my .cpproj where it gives me these errors:

1. After running `pipeline.load("CFReT_project.cpproj")`:
```
WARNING:root:MySQL could not be loaded.
Traceback (most recent call last):
  File "/Users/jennatomkinson/opt/anaconda3/envs/cp4/lib/python3.8/site-packages/cellprofiler/modules/exporttodatabase.py", line 154, in <module>
    import MySQLdb
  File "/Users/jennatomkinson/opt/anaconda3/envs/cp4/lib/python3.8/site-packages/MySQLdb/__init__.py", line 18, in <module>
    from . import _mysql
ImportError: dlopen(/Users/jennatomkinson/opt/anaconda3/envs/cp4/lib/python3.8/site-packages/MySQLdb/_mysql.cpython-38-darwin.so, 0x0002): Library not loaded: '@rpath/libmysqlclient.20.dylib'
  Referenced from: '/Users/jennatomkinson/opt/anaconda3/envs/cp4/lib/python3.8/site-packages/MySQLdb/_mysql.cpython-38-darwin.so'
  Reason: tried: '/Users/jennatomkinson/opt/anaconda3/envs/cp4/bin/../lib/libmysqlclient.20.dylib' (no such file), '/Users/jennatomkinson/opt/anaconda3/envs/cp4/bin/../lib/libmysqlclient.20.dylib' (no such file), '/usr/local/lib/libmysqlclient.20.dylib' (no such file), '/usr/lib/libmysqlclient.20.dylib' (no such file)
```

2. After running `output = pipeline.run()`:
```
ERROR:root:Failed to prepare run for module ExportToDatabase
Traceback (most recent call last):
  File "/Users/jennatomkinson/opt/anaconda3/envs/cp4/lib/python3.8/site-packages/cellprofiler/modules/exporttodatabase.py", line 345, in get_next_result
    return next(cursor)
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/jennatomkinson/opt/anaconda3/envs/cp4/lib/python3.8/site-packages/cellprofiler/modules/exporttodatabase.py", line 2457, in prepare_run
    self.create_database_tables(self.cursor, workspace)
  File "/Users/jennatomkinson/opt/anaconda3/envs/cp4/lib/python3.8/site-packages/cellprofiler/modules/exporttodatabase.py", line 3101, in create_database_tables
    execute(
  File "/Users/jennatomkinson/opt/anaconda3/envs/cp4/lib/python3.8/site-packages/cellprofiler/modules/exporttodatabase.py", line 331, in execute
    return get_results_as_list(cursor)
  File "/Users/jennatomkinson/opt/anaconda3/envs/cp4/lib/python3.8/site-packages/cellprofiler/modules/exporttodatabase.py", line 335, in get_results_as_list
    r = get_next_result(cursor)
  File "/Users/jennatomkinson/opt/anaconda3/envs/cp4/lib/python3.8/site-packages/cellprofiler/modules/exporttodatabase.py", line 346, in get_next_result
    except MySQLdb.Error as e:
NameError: name 'MySQLdb' is not defined

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/jennatomkinson/Desktop/Github/CellProfiler/core/cellprofiler_core/pipeline/_pipeline.py", line 1432, in prepare_run
    not module.prepare_run(workspace)
  File "/Users/jennatomkinson/opt/anaconda3/envs/cp4/lib/python3.8/site-packages/cellprofiler/modules/exporttodatabase.py", line 2459, in prepare_run
    except sqlite3.OperationalError as err:
NameError: name 'sqlite3' is not defined
```
It then outputs an empty .sqlite file. Any advice on how to solve this issue is much appreciated. Thank you!

**To Reproduce**
1. Follow the steps in the [CellProfiler as a Python package](https://github.com/CellProfiler/CellProfiler/wiki/CellProfiler-as-a-Python-package) page up to the run step.
2. Run the pipeline (an example of a pipeline like the one I used is located [here](https://github.com/WayScience/NF1_SchwannCell_data/blob/main/CellProfiler_pipelines/Pipelines/NF1_analysis.cpproj). It just needs to have the `ExporttoDatabase` module for this error to happen.

**Expected behavior**
The expected behavior is for the .cpproj file to not be able to run, display the errors above and it will output a blank .sqlite file.

**Screenshots**
Error 1:
<img width="1440" alt="Screen Shot 2022-09-22 at 4 00 48 PM" src="https://user-images.githubusercontent.com/107513215/191859226-c4270614-a1b1-47aa-bdd1-0265ff9950e2.png">

Error 2: Please do not mind the warnings, the dataset is missing images and I working on that.
<img width="1440" alt="Screen Shot 2022-09-22 at 4 01 12 PM" src="https://user-images.githubusercontent.com/107513215/191859390-f49c0307-1ddc-4a77-a5da-7be4ac7f304d.png">

Here is what my CellProfiler pipeline looks like in the GUI:
<img width="1096" alt="Screen Shot 2022-09-22 at 4 07 10 PM" src="https://user-images.githubusercontent.com/107513215/191859988-e3c4eea7-045b-478e-8b9f-c08c96af2d9b.png">

**Desktop (please complete the following information):**
 - OS: macOS Monterey version 12.5.1
 - CellProfiler Version: 4.2.4
 - Conda environment created using from source

**Comment by bethac07** [comment]
Sorry to hear this - a couple of questions

1) Have you made sure to install all the [underlying requirements](https://github.com/CellProfiler/CellProfiler/wiki/Source-installation-%28OS-X-and-macOS%29), like mysql?
2) Do you get the same error when you use a cppipe file rather than a cpproj file? I don't think cpproj files are meant to be used in this way.
3) Do you need to run CellProfiler as a Python package? Realistically, there are not very many good use cases for doing it in this fashion that will be easier than just running things in sequence or [running CellProfiler via its existing command line interface](https://github.com/CellProfiler/CellProfiler/wiki/Getting-started-using-CellProfiler-from-the-command-line) with something like `subprocess`. Running it as a package is not recommended in most contexts.

**Comment by jenna-tomkinson** [comment]
Hi Beth!

1. I do believe so, but I will double check. I have learned that what I assume was installed isn't always right haha.
2. I have not been exporting the pipelines as .cppipe so I will test that and update you. I actually used the Fruit Fly example pipeline from the example and opened it in CP 4.2.4 GUI, then saved it as a .cpproj to test if it is the file type that is the issue. I does create different warnings but still runs fine. 
3. I am now testing using the command line. Hopefully that works better! 

Thank you for your help!
